cmd/controller/pkg_repository_reconciler.go

```diff
@@ -66,6 +75,68 @@ func (r *PkgRepositoryReconciler) Reconcile(ctx context.Context, request reconci
 	return crdApp.Reconcile(force)
 }
 
+// ReconcileFetchPlaceholderSecrets helps determine if a placeholder secret
+// needs to be created for the PackageRepository. This placeholder secret is
+// populated by secretgen-controller so PackageRepositories can authenticate
+// to private registries without needing to explicitly declare a secretRef.
+// If no secretRef is specified for the PackageRepository, the placeholder
+// is created and used by the PackageRepository.
+func (r *PkgRepositoryReconciler) ReconcileFetchPlaceholderSecrets(pkgr v1alpha12.PackageRepository, app *v1alpha1.App) error {
```

Comment:
not ideal that this is publicly exposed on the reconciler, but it's fine for now i think, especially that package repo is implemented via internal App. might be good to play around with some refactoring post release.

Reply:
This is mainly done for testing purposes with the function. Alternatively, the tests could be under the controller package as opposed to controller_test.

Reply:
yup. just noting. controller_test is good.